code.js

```diff
@@ -1,3 +1,26 @@
-function permutationSort(a) {
-    return 0;
+function permutationSort(a, q = [], iters = 0, len = a.length, done = [false]) {
```

Comment:
Why is `done` an array?

Reply:
So it can be passed by reference through the recursive calls.

Reply:
Hmm, unless I'm missing something you'd only need to do that if you were doing the recursive calls in parallel. It should work if you just use a Boolean instead of the array.